### DISCUSSION TITLE
How to change value type after type check

### DISCUSSION THREAD

**Discussion by zjaper456** [python/typing#1294] [Q&A]
my code
``` python
_, ata_ret, id_data = ata.identify_device()
check.not_equal(ata_ret, None)
check.equal(ata_ret.status & 0x1, 0)
```
the ata_ret type is `Union[ATAReturn, FixedATAReturn, None]`

``` python
@staticmethod
def not_equal(value: Any, expect: Any) -> None:
    assert value != expect
```

i got a error info `Item "None" of "Union[ATAReturn, FixedATAReturn, None]" has no attribute "status"mypy(error)`

how can i fix it?

**Comment by jeremyn** [comment]
Try using the [TypeGuard](https://peps.python.org/pep-0647/) feature, particularly the "more than one argument" variant described in the Specification section of the linked PEP.
